src/extra/components/pulses.py

```diff
@@ -940,15 +959,111 @@ def _identify_ppl_seed(cls, data):
     def _mask_table(self, table):
         return is_laser(table, self._ppl_seed)
 
-    def _get_ppdecoder_node(self):
-        return 'laser'
+    def _get_ppdecoder_nodes(self):
+        return ['laser']
 
     @property
     def ppl_seed(self) -> Optional[PPL_BITS]:
         """Used laser seed."""
         return self._ppl_seed
 
 
+class MachinePulses(TimeserverPulses):
+    """An interface to machine pulses and other bunch pattern data.
+
+    The bunch pattern table contains many more information than the
```

Comment:
Nitpicking:
```suggestion
    The bunch pattern table contains much more information than just the
```

Reply:
That also sounds quite clunky to me. How about we switch to a boolean flag like `require_all_bits=False`?

Reply:
Or, if it's more intuitive, `require_any_bit=True`

Reply:
Sure, both of those are quite readable.